### PR TITLE
[common/dv] common_ifs documentation update

### PR DESCRIPTION
**Pull request by gaurangchitroda**
common_ifs documentation update

### CODE REVIEW

File: hw/dv/sv/common_ifs/README.md [hunk 1]
```diff
@@ -3,11 +3,63 @@
 {{% toc 4 }}
 
 ## Overview
-In this directory, we provide commonly used interfaces to construct testbenches for DV.
```
[the remark this answers is not in the document]
Comment (gaurangchitroda) [reply hw/dv/sv/common_ifs/README.md]: @sriyerg @gkelly - I am going to close this PR and open a new one by making a new commit. I shall mention a reference to this PR.

[the remark this answers is not in the document]
Comment (gkelly) [reply hw/dv/sv/common_ifs/README.md]: A link to `{{< relref "hw/ip/some_document" >}}` where `some_document.md` exists doesn't need to be updated if it's moved to `some_document/index.md`.

**Comment by gaurangchitroda** [comment]
Please refer to new PR #896 for further commit updates

File: hw/dv/sv/common_ifs/README.md [hunk 2]
```diff
@@ -3,11 +3,63 @@
...
-They are described in detail below.
+In this directory, we provide commonly used interfaces used to construct
+testbenches for DV. These interfaces are instantiated inside `tb` module for
+connecting dut signals. They are described in detail below.
 
-### clk_rst_if
+### `clk_if`
+This is a generic clock interface used generating clock events in testbenches.
```
[the remark this answers is not in the document]
Comment (gaurangchitroda) [reply hw/dv/sv/common_ifs/README.md]: Fixed this in new commit with #896

File: hw/dv/sv/common_ifs/README.md [hunk 3]
```diff
@@ -3,11 +3,63 @@
...
+This interface has two clocking blocks: `cb` and `cbn` for synchronizing to
+positive and negative clock edges. This interface consists of following tasks:
+* `wait_clks`: waits for specified number of positive clock edges
+* `wait_n_clks`: waits for specified number of negative clock edges
+
+### `clk_rst_if`
```
[the remark this answers is not in the document]
Comment (gaurangchitroda) [reply hw/dv/sv/common_ifs/README.md]: I have updated description for this in #896

File: hw/dv/sv/common_ifs/README.md [hunk 4]
```diff
@@ -3,11 +3,63 @@
...
+output port within of dut to drive or sample them. `pins` can be driven either
+internally using `pins_o` and `pins_oe` signals, that constitute a tri-state
+buffer implementation. This provide an ability to disconnects `pins` by driving
+them to high impedance state. `pins` may also be driven through an external
+driver that it gets connected to. This interface also provides capability
+to drive weak pull-up or pull-down on `pins` in case of no internal or external
+drivers. The members `pins_pu` and `pins_pd` control weak pull-up or pull-down
+functionality. Following diagram explains working of `pins_if`:
+
+![Block diagram](pins_if.svg)
```
[the remark this answers is not in the document]
Comment (gaurangchitroda) [reply hw/dv/sv/common_ifs/README.md]: Addressed this in #896

File: hw/dv/sv/common_ifs/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (gaurangchitroda) [reply hw/dv/sv/common_ifs/README.md]: As per discussion with @sriyerg , I have renamed `common_ifs/README.md` to `common_ifs/index.md` and also updated references to this file in other markdown files. Updates submitted as a part of #896.

**Pull request closed without merging**